modules/imgcodecs/src/grfmt_hdr.cpp

```diff
@@ -141,14 +141,28 @@ bool HdrEncoder::write( const Mat& input_img, const std::vector<int>& params )
     if(img.depth() != CV_32F) {
         img.convertTo(img, CV_32FC3, 1/255.0f);
     }
-    CV_Assert(params.empty() || params[0] == HDR_NONE || params[0] == HDR_RLE);
+
+    int compression = IMWRITE_HDR_COMPRESSION_RLE;
+    for (size_t i = 0; i + 1 < params.size(); i += 2)
+    {
+        switch (params[i])
+        {
+        case CV_IMWRITE_JPEG_QUALITY:
```

Comment:
I feel that it is need to fix the conditional here.

```c
-     case CV_IMWRITE_JPEG_QUALITY:
+     case IMWRITE_HDR_COMPRESSION
```

Reply:
@Kumataro Thank you! Fixed.
Also fixed `imencode()` and added tests.

BTW, C API is on hold. We don't modify it anymore.

Reply:
Thank you to fix it , and I agree with you. 

`CV_IMWRITE_HDR_COMPRESSION‘  is not necessary and will not be defined.